File: app/src/main/java/se/curtrune/lucy/workers/MentalWorker.java
package se.curtrune.lucy.workers;

import static se.curtrune.lucy.util.Logger.log;

import android.content.Context;

import java.time.LocalDate;
import java.util.List;

import se.curtrune.lucy.classes.item.Item;
import se.curtrune.lucy.classes.Mental;
import se.curtrune.lucy.classes.State;
import se.curtrune.lucy.persist.ItemsWorker;
import se.curtrune.lucy.persist.SqliteLocalDB;
import se.curtrune.lucy.persist.Queeries;

public class MentalWorker {
    public static boolean VERBOSE = false;
    
    /*** @param date the date you wish to examine
     * @param context, context
     * @return a Mental representing the sum of all the item said date
     */
    public static Mental getMental(LocalDate date, Context context) {
        log("MentalWorker.getMental(LocalDate, Context)");
        List<Item> items = ItemsWorker.selectItems(date, context);
        int anxiety = 0, energy = 0, mood = 0, stress = 0;
        anxiety = items.stream().mapToInt(Item::getAnxiety).sum();
        energy = items.stream().mapToInt(Item::getEnergy).sum();
        mood = items.stream().mapToInt(Item::getMood).sum();
        stress = items.stream().mapToInt(Item::getStress).sum();
        return new Mental(anxiety, energy, mood, stress);
    }

    /**
     *
     * @param date the date to interrogate
     * @param context, context
     * @return Mental sum for done items
     */
    public static Mental getCurrentMental(LocalDate date, Context context) {
        log("MentalWorker.getCurrentMental(LocalDate)", date.toString());
        String queery = Queeries.selectItems(date, State.DONE);
        List<Item> items;
        int anxiety = 0;
        int energy = 0;
        int mood = 0;
        int stress = 0;
        try(SqliteLocalDB db = new SqliteLocalDB(context)){
            items = db.selectItems(queery);
            anxiety = items.stream().mapToInt(Item::getAnxiety).sum();
            energy = items.stream().mapToInt(Item::getEnergy).sum();
            mood = items.stream().mapToInt(Item::getMood).sum();
            stress = items.stream().mapToInt(Item::getStress).sum();
        }catch (Exception exception){
            exception.printStackTrace();
            log(exception.getMessage());
        }
        return new Mental(anxiety, energy, mood, stress);
    }
}

File: app/src/main/java/se/curtrune/lucy/workers/MentalWorker.kt
package se.curtrune.lucy.workers

import android.content.Context
import se.curtrune.lucy.modules.LucindaApplication
import se.curtrune.lucy.classes.Mental
import se.curtrune.lucy.classes.State
import se.curtrune.lucy.classes.item.Item
import se.curtrune.lucy.persist.Queeries
import se.curtrune.lucy.persist.SqliteLocalDB
import se.curtrune.lucy.util.Logger
import java.time.LocalDate

object MentalWorker {
    var VERBOSE: Boolean = false
    val repository = LucindaApplication.appModule.repository

    /*** @param date the date you wish to examine
     * @param context, context
     * @return a Mental representing the sum of all the item said date
     */
    fun getMental(date: LocalDate, context: Context?): Mental {
        Logger.log("MentalWorker.getMental(LocalDate, Context)")
        val items: List<Item> = repository.selectItems(date)
        var anxiety = 0
        var energy = 0
        var mood = 0
        var stress = 0
        anxiety = items.stream().mapToInt { obj: Item -> obj.anxiety }.sum()
        energy = items.stream().mapToInt { obj: Item -> obj.energy }.sum()
        mood = items.stream().mapToInt { obj: Item -> obj.mood }.sum()
        stress = items.stream().mapToInt { obj: Item -> obj.stress }.sum()
        return Mental(anxiety, energy, mood, stress)
    }
    fun getMental(items: List<Item>): Mental {
        println("MentalWorker.getMental(List<Item>)")
        val mental = Mental()
        mental.anxiety = items.stream().mapToInt { obj: Item -> obj.anxiety }.sum()
        mental.energy = items.stream().mapToInt { obj: Item -> obj.energy }.sum()
        mental.mood = items.stream().mapToInt { obj: Item -> obj.mood }.sum()
        mental.stress = items.stream().mapToInt { obj: Item -> obj.stress }.sum()
        return mental
    }

    /**
     *
     * @param date the date to interrogate
     * @param context, context
     * @return Mental sum for done items
     */
    fun getCurrentMental(date: LocalDate, context: Context?): Mental {
        println("MentalWorker.getCurrentMental(LocalDate $date)")
        val queery = Queeries.selectItems(date, State.DONE)
        var items: List<Item>
        var anxiety = 0
        var energy = 0
        var mood = 0
        var stress = 0
        try {
            SqliteLocalDB(context).use { db ->
                items = db.selectItems(queery)
                anxiety = items.stream().mapToInt { obj: Item -> obj.anxiety }.sum()
                energy = items.stream().mapToInt { obj: Item -> obj.energy }.sum()
                mood = items.stream().mapToInt { obj: Item -> obj.mood }.sum()
                stress = items.stream().mapToInt { obj: Item -> obj.stress }.sum()
            }
        } catch (exception: Exception) {
            exception.printStackTrace()
            Logger.log(exception.message)
        }
        return Mental(anxiety, energy, mood, stress)
    }
}
